src/modules/json.wren

```diff
@@ -0,0 +1,245 @@
+class JsonError {
+    line { _line }
+    position { _position }
+    message { _message }
+    found { _found }
+
+    construct new(line, pos, message, found) {
+        _line = line
+        _position = pos
+        _message = message
+        _found = found
+    }
+
+    static empty() {
+        return JsonError.new(0, 0, "", false)
+    }
+}
+
+class JsonStream {
+    foreign stream_begin(value)
+    foreign stream_end()
+    foreign next
+    foreign value
+    foreign error_message
+    foreign lineno
+    foreign pos
+    foreign static escapechar(value, options)
+
+    namespace { "com.domeengine.json:" }
+    isNull { namespace + "JSON_NULL" }
+    isString { namespace + "JSON_STRING" }
+    isNumeric { namespace + "JSON_NUMBER" }
+    isBoolTrue { namespace + "JSON_TRUE" }
+    isBoolFalse { namespace + "JSON_FALSE" }
+    isArray { namespace + "JSON_ARRAY" }
+    isArrayEnd { namespace + "JSON_ARRAY_END" }
+    isObject { namespace + "JSON_OBJECT" }
+    isObjectEnd { namespace + "JSON_OBJECT_END" }
+    isDone { namespace + "JSON_DONE" }
+    isError { namespace + "JSON_ERROR" }
+    isInit { namespace + "JSON_INIT" }
+
+    result { _result }
+    error { _error }
+    options { _options }
+    raw { _raw }
+
+    construct new(raw, options) {
+        _result = {}
+        _error = JsonError.empty()
+        _lastEvent = isInit
+        _raw = raw
+        _options = options
+    }
+
+    begin() {
+        stream_begin(_raw)
+        _result = process(next)
+    }
+
+    end() {
+        stream_end()
+    }
+
+    process(event) {
+
+        _lastEvent = event
+
+        if (event == isBoolTrue || event == isBoolFalse) {
+            return (event == isBoolTrue)
+        }
+
+        if (event == isNumeric) {
+            return Num.fromString(this.value)
+        }
+
+        if (event == isString) {
+            return this.value
+        }
+
+        if (event == isNull) {
+            return null
+        }
+
+        if (event == isArray) {
+            var elements = []
+            while (true) {
+                event = next
+                _lastEvent = event
+                if (event == isArrayEnd || event == isDone || event == isError) {
+                    break
+                }
+                elements.add(process(event))
+            }
+            return elements
+        }
+
+        if (event == isObject) {
+            var elements = {}
+            while (true) {
+                event = next
+                _lastEvent = event
+                if (event == isObjectEnd || event == isDone || event == isError) {
+                    break
+                }
+                elements[this.value] = process(next)
+            }
+            return elements
+        }
+
+        if (event == isError) {
+            _error = JsonError.new(lineno, pos, error_message, true)
+            if (JsonOptions.shouldAbort(_options)) {
+                Fiber.abort("JSON Error: line %(lineno) pos %(pos): %(error_message)")
```

Comment:
Please change this from `JSON Error: ` to `JSON error - ` so that the Runtime Error output looks a bit less repetitive.

Reply:
ok